tests/_test_msui/test_mscolab.py

```diff
@@ -59,7 +59,7 @@ def setup(self, qtbot, mscolab_server):
         assert add_user_to_operation(path=self.operation_name, emailid=self.userdata[0])
         self.user = get_user(self.userdata[0])
 
-        self.main_window = msui.MSUIMainWindow(mscolab_data_dir=mscolab_settings.MSCOLAB_DATA_DIR)
+        self.main_window = msui.MSUIMainWindow(local_operations_data=ROOT_DIR)
```

Comment:
Is reusing ROOT_DIR a good idea, or should it better be a sub-directory (or even an entirely different temporary directory)?

Reply:
ROOT_DIR is for now ok. On tests that is a tmp dir.

Reply:
We have currently not an own name for this async data dir. That is hardcoded. With the changes now all data is below the ROOT_DIR in this case ~/mss. 
The current setup of the dirs for the asynchronous data (local_colabdata) take care for the tests or developers using different accounts.
The setup on a developers machine of an mscolab server uses colabdata. There each dir is a git repo, while in local_colabdata we have only 1 file in recent version


```
~/mss$ tree
.
├── colabdata
│   ├── datasso
│   ├── filedata
│   │   ├── Admin_Test
│   │   │   └── main.ftml
│   │   ├── four
│   │   │   └── main.ftml
│   │   ├── one
│   │   │   └── main.ftml
│   │   ├── test_mscolab
│   │   │   └── main.ftml
│   │   ├── three
│   │   │   └── main.ftml
│   │   └── two
│   │       └── main.ftml
│   ├── mscolab.db
│   └── uploads
└── local_colabdata
    └── a
        └── one
            └── mscolab_operation.ftml
```

Reply:
So essentially, MSUI only ever writes to `$ROOT_DIR/local_colabdata`? If that is the case, then there shouldn't be any conflicts by using it, but I still think that it is passed the wrong directory then if it needs to suffix it with another directory unconditionally.

I.e. I think it should be passed `local_operations_data=(ROOT_DIR / "local_colabdata")` (here in pathlib notation, don't know what it would really need to look like right now) since that is where local operations data is stored, and don't append another directory itself.

This could be a future refactor though, as not reusing mscolab_settings was already tangential to this PR. I just want to find out if my understanding of what it does is somewhat correct.

Reply:
additional the word "root dir" does come from the pyfilesystem2 idea. e.g. https://docs.pyfilesystem.org/en/latest/concepts.html?highlight=root#sandboxing

That wording needs then also refactored when do https://github.com/Open-MSS/MSS/issues/2103

Reply:
> So essentially, MSUI only ever writes to `$ROOT_DIR/local_colabdata`? If that is the case, then there shouldn't be any conflicts by using it, but I still think that it is passed the wrong directory then if it needs to suffix it with another directory unconditionally.
> 
> I.e. I think it should be passed `local_operations_data=(ROOT_DIR / "local_colabdata")` (here in pathlib notation, don't know what it would really need to look like right now) since that is where local operations data is stored, and don't append another directory itself.
> 
> This could be a future refactor though, as not reusing mscolab_settings was already tangential to this PR. I just want to find out if my understanding of what it does is somewhat correct.

MSUI writes also ftml, csv, etc. files when mscolab is not used. They have a different dir.  We should do a different refactoring (not here) to use the same base dir for both. This would be the first step to reuse the mscolab async files in a diffferent way.

Reply:
I am currently also not sure if we have to keep all levels. user-operation-file(with same name: mscolab_operation.ftml)

Reply:
https://github.com/Open-MSS/MSS/issues/2508